### DISCUSSION TITLE
OpenDDS in SpringBoot

### DISCUSSION THREAD

**Discussion by Alwaysnopro** [OpenDDS/OpenDDS#3109] [Q&A] [answered]
When I start a new springboot project, I use the publisher.java as a publisher, forthermore, I create a web page with a button which could be used to execute publisher.java. The subscriber running in another machine.
When I click the button firstly, the subscriber could receive messages. however, an error occurred during the button being clicked secondly.
Controller
    @RequestMapping("/DTCBPost")
    @ResponseBody
    public Result Trans(DTCB dtcb){
        Publisher publisher = new Publisher();
        list.add(dtcb);
        publisher.publisher(args,list);
}

publisher.java
    public void publisher(String[] args, List<DTCB> list) {

        System.out.println("Start Publisher");
        boolean reliable = checkReliable(args);
        boolean waitForAcks = checkWaitForAcks(args);

        DomainParticipantFactory dpf =
            TheParticipantFactory.WithArgs(new StringSeqHolder(args));
        if (dpf == null) {
            System.err.println("ERROR: Domain Participant Factory not found");
            return;
        }
(8628|16440) ERROR: Service_Participant::load_configuration open() returned -1
(8628|16440) ERROR: Service_Participant::get_domain_participant_factory: load_configuration() failed.
ERROR: Domain Participant Factory not found

**Comment by mitza-oci** [comment, marked as answer]
The setup steps (creating a DomainParticipantFactory, etc.) should happen when the application initializes, not when handling an individual POST request.  The error is probably that it's already initialized and so shouldn't be initialized again unless shut down first.
